src/uwtools/drivers/driver.py

```diff
@@ -154,6 +123,133 @@ def _driver_name(self) -> str:
         Returns the name of this driver.
         """
 
+    @property
+    def _resources(self) -> Dict[str, Any]:
+        """
+        Returns configuration data for the runscript.
+        """
+        try:
+            platform = self._config["platform"]
+        except KeyError as e:
+            raise UWConfigError("Required 'platform' block missing in config") from e
+        return {
+            "account": platform["account"],
+            "rundir": self._rundir,
+        }
+
+    @property
+    def _rundir(self) -> Path:
+        """
+        The path to the component's run directory.
+        """
+        return Path(self._driver_config["run_dir"])
+
+    def _taskname(self, suffix: str) -> str:
+        """
+        Returns a common tag for graph-task log messages.
+
+        :param suffix: Log-string suffix.
+        """
+        return "%s %s" % (self._driver_name, suffix)
+
+    def _taskname_with_cycle(self, cycle: datetime, suffix: str) -> str:
+        """
+        Returns a common tag for graph-task log messages.
+
+        :param suffix: Log-string suffix.
+        """
+        return "%s %s %s" % (cycle.strftime("%Y%m%d %HZ"), self._driver_name, suffix)
+
+    def _taskname_with_cycle_and_leadtime(
+        self, cycle: datetime, leadtime: timedelta, suffix: str
+    ) -> str:
+        """
+        Returns a common tag for graph-task log messages.
+
+        :param suffix: Log-string suffix.
+        """
+        return "%s %s %s" % (
+            (cycle + leadtime).strftime("%Y%m%d %H:%M:%S"),
+            self._driver_name,
+            suffix,
+        )
+
+    def _validate(self) -> None:
+        """
+        Perform all necessary schema validation.
+        """
+        for schema_name in (self._driver_name.replace("_", "-"), "platform"):
+            validate_internal(schema_name=schema_name, config=self._config)
+
+
+class StandaloneDriver(Driver):
+    """
+    An abstract class for component drivers.
```

Comment:
Whatever class names we settle on, probably this class and its parent should have different docstrings to make it clearer why they are separate classes.

Reply:
True! This may have gotten lost in revisions.
Also, offline discussion prefers `Driver(Assets)` and `ComponentDriver(ComponentAssets)` for naming so I'll be implementing it